util/stream.go

```diff
@@ -149,23 +179,14 @@ func (m *MessageStream) inbound() {
 	}
 }
 
-// Parse incoming message
-func (m *MessageStream) parse() {
-	errMessage := "received: %v and encountered error: %v"
-	for {
-		select {
-		case b := <-m.pool.Full:
-			msg, err := m.parser.Parse(b.Bytes())
-			// Log all message parsing errors.
-			if err != nil {
-				log.Errorf(errMessage, b.Bytes(), err)
-			}
-
-			m.Inbound <- msg
-			b.Reset()
-			m.pool.Empty <- b
-		case <-m.parserShutdown:
-			return
-		}
+// Dispatch the message to streamWorker according to Xid in the message Header
+func (m *MessageStream) dispatchMessage(b *bytes.Buffer) {
+	msgBytes := b.Bytes()
+	if len(msgBytes) < 8 {
+		log.Error("bytes too small to pares OpenFlow messages")
```

Comment:
typo: s/pares/parse

I think it should be `"buffer too small to parse OpenFlow messages"` for clarity.

Also, just to confirm: this shouldn't happen since all the allocated buffers have a capacity of 2048 bytes?

Reply:
Updated the log.
This should not happen in theory, and this check here is for security.